src/dataflow/render/mod.rs

```diff
@@ -983,528 +1003,17 @@ where
-                offset += arities[input];
-            }
-            // let input_relation = arities
-            //     .iter()
-            //     .enumerate()
-            //     .flat_map(|(r, a)| std::iter::repeat(r).take(*a))
-            //     .collect::<Vec<_>>();
-
-            // Unwrap demand
-            // TODO: If we pushed predicates into the operator, we could have a
-            // more accurate view of demand that does not include the support of
-            // all predicates.
-            let demand = demand.clone().unwrap_or_else(|| (0..arity).collect());
-
-            // This collection will evolve as we join in more inputs.
-            let (mut joined, mut errs) = self.collection(&inputs[*start]).unwrap();
-
-            // Maintain sources of each in-progress column.
-            let mut source_columns = (prior_arities[*start]
-                ..prior_arities[*start] + arities[*start])
-                .collect::<Vec<_>>();
-
-            let mut predicates = predicates.to_vec();
-            let (j, es) = crate::render::delta_join::build_filter(
-                joined,
-                &source_columns,
-                &mut predicates,
-                &mut equivalences,
-            );
-            joined = j;
-            if let Some(es) = es {
-                errs.concat(&es);
-            }
-
-            for (input, next_keys) in order.iter() {
-                let mut next_keys_rebased = next_keys.clone();
-                for expr in next_keys_rebased.iter_mut() {
-                    expr.visit_mut(&mut |e| {
-                        if let ScalarExpr::Column(c) = e {
-                            *c += prior_arities[*input];
-                        }
-                    });
-                }
-
-                // Keys for the next input to be joined must be produced from
-                // ScalarExprs found in `equivalences`, re-written to bind the
-                // appropriate columns (as `joined` has permuted columns).
-                let prev_keys = next_keys_rebased
-                    .iter()
-                    .map(|expr| {
-                        // We expect to find `expr` in some `equivalence` which
-                        // has a bound expression. Otherwise, the join plan is
-                        // defective and we should panic.
-                        let equivalence = equivalences
-                            .iter()
-                            .find(|equivs| equivs.contains(expr))
-                            .expect("Expression in join plan is not in an equivalence relation");
-
-                        // We expect to find exactly one bound expression, as
-                        // multiple bound expressions should result in a filter
-                        // and be removed once they have.
-                        let mut bound_expr = equivalence
+                RelationExpr::ArrangeBy { input, keys } => {
+                    if keys.is_empty()
```

Comment:
Can you add a comment saying we're not really sure this is necessary and just maintaining the old behavior

Reply:
It is currently required based on the docs for `RelationExpr::ArrangeBy` which says that it is a no-op (and by my reading, should ensure that `input` is available independent of its arguments). I'll add an explanation.